src/components/PostsList/PostsList.tsx

```diff
@@ -1,37 +1,55 @@
 import React from 'react';
 import './PostsList.scss';
 
-export const PostsList: React.FC = () => (
-  <div className="PostsList">
-    <h2>Posts:</h2>
+type Props = {
+  selectedPostId: number | null;
+  setSelectedPostId: (id: number) => void;
+  posts: Post[] | null;
+};
 
-    <ul className="PostsList__list">
-      <li className="PostsList__item">
-        <div>
-          <b>[User #1]: </b>
-          sunt aut facere repellat provident occaecati excepturi optio
-        </div>
-        <button
-          type="button"
-          className="PostsList__button button"
-        >
-          Close
-        </button>
-      </li>
+export const PostsList: React.FC<Props> = ({
+  selectedPostId,
+  setSelectedPostId,
+  posts,
+}) => {
+  return (
+    <div className="PostsList">
+      <h2>Posts:</h2>
 
-      <li className="PostsList__item">
-        <div>
-          <b>[User #2]: </b>
-          et ea vero quia laudantium autem
-        </div>
-
-        <button
-          type="button"
-          className="PostsList__button button"
-        >
-          Open
-        </button>
-      </li>
-    </ul>
-  </div>
-);
+      <ul className="PostsList__list">
+        {posts?.map(post => (
+          <li
+            key={post.id}
+            className="PostsList__item"
+          >
+            <div>
+              <b>{`[User #${post.userId}]: `}</b>
+              {post.title}
+            </div>
+            {selectedPostId === post.id
+              ? (
+                <button
+                  type="button"
+                  className="
+                    PostsList__button
+                    button"
+                  onClick={() => setSelectedPostId(0)}
+                >
+                  Close
+                </button>
+              )
+              : (
+                <button
+                  type="button"
+                  className="PostsList__button button"
+                  onClick={() => setSelectedPostId(post.id)}
+                >
+                  Open
+                </button>
+              )}
```

Comment:
it is the same button, despite text and function argument, so why do we need to write it twice